The job **`aarch64-gnu-llvm-20-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/20740785373/job/59547054550) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/59547054550) [(plain)](https://github.com/rust-lang/rust/commit/507b29c3eba101015339710aac4de1d0cb659455/checks/59547054550/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

---- [codegen] tests/codegen-llvm/lib-optimizations/append-elements.rs stdout ----
------FileCheck stdout------------------------------

------FileCheck stderr------------------------------
/checkout/tests/codegen-llvm/lib-optimizations/append-elements.rs:12:12: error: CHECK: expected string not found in input
 // CHECK: %[[TEMP:.+]] = tail call noalias noundef ptr @{{.*}}__rust_alloc
           ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/lib-optimizations/append-elements/append-elements.ll:124:40: note: scanning from here
define void @vec_append_with_temp_alloc(ptr noalias nocapture noundef align 8 dereferenceable(24) %dst, ptr noalias nocapture noundef nonnull readonly align 1 %src.0, i64 noundef range(i64 0, -9223372036854775808) %src.1) unnamed_addr #1 personality ptr @rust_eh_personality {
                                       ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/lib-optimizations/append-elements/append-elements.ll:132:1: note: possible intended match here
; call __rustc::__rust_alloc
^
/checkout/tests/codegen-llvm/lib-optimizations/append-elements.rs:27:12: error: CHECK: expected string not found in input
 // CHECK: %[[TEMP:.+]] = tail call noalias noundef ptr @__rust_alloc
           ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/lib-optimizations/append-elements/append-elements.ll:191:43: note: scanning from here
define void @string_append_with_temp_alloc(ptr noalias nocapture noundef align 8 dereferenceable(24) %dst, ptr noalias nocapture noundef nonnull readonly align 1 %src.0, i64 noundef %src.1) unnamed_addr #1 personality ptr @rust_eh_personality {
                                          ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/lib-optimizations/append-elements/append-elements.ll:199:1: note: possible intended match here
; call __rustc::__rust_alloc
^

Input file: /checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/lib-optimizations/append-elements/append-elements.ll
Check file: /checkout/tests/codegen-llvm/lib-optimizations/append-elements.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
            .
            .
            .
           24:  %_0.sroa.0.0.sroa.speculated.i.i = tail call noundef i64 @llvm.umax.i64(i64 %_28.0.i, i64 range(i64 0, -1) %1) 
           25:  %_0.sroa.0.0.sroa.speculated.i15.i = tail call noundef i64 @llvm.umax.i64(i64 %_0.sroa.0.0.sroa.speculated.i.i, i64 8) 
           26:  call void @llvm.lifetime.start.p0(i64 24, ptr nonnull %self3.i), !noalias !2 
           27:  %self.val.i = load ptr, ptr %slf, align 8, !alias.scope !2 
           28: ; call <alloc::raw_vec::RawVecInner>::finish_grow 
           29:  call fastcc void @_RNvMs2_NtCsj7oXjtrRIDu_5alloc7raw_vecNtB5_11RawVecInner11finish_growCslAzzQRh6qCt_15append_elements(ptr noalias noundef align 8 dereferenceable(24) %self3.i, ptr %self.val.i, i64 %self5.i, i64 noundef %_0.sroa.0.0.sroa.speculated.i15.i) 
           30:  %_38.i = load i64, ptr %self3.i, align 8, !range !7, !noalias !2, !noundef !6 
           31:  %2 = trunc nuw i64 %_38.i to i1 
           32:  %3 = getelementptr inbounds nuw i8, ptr %self3.i, i64 8 
           33:  br i1 %2, label %bb21.i, label %bb3 
           34:  
           35: bb21.i: ; preds = %bb12.i 
           36:  %e.0.i = load i64, ptr %3, align 8, !range !8, !noalias !2, !noundef !6 
           37:  %4 = getelementptr inbounds nuw i8, ptr %self3.i, i64 16 
           38:  %e.1.i = load i64, ptr %4, align 8, !noalias !2 
           39:  call void @llvm.lifetime.end.p0(i64 24, ptr nonnull %self3.i), !noalias !2 
           40:  br label %bb2 
           41:  
           42: bb2: ; preds = %bb21.i, %start 
           43:  %_0.sroa.5.0.i.ph = phi i64 [ undef, %start ], [ %e.1.i, %bb21.i ] 
           44:  %_0.sroa.0.0.i.ph = phi i64 [ 0, %start ], [ %e.0.i, %bb21.i ] 
           45: ; call alloc::raw_vec::handle_error 
           46:  tail call void @_RNvNtCsj7oXjtrRIDu_5alloc7raw_vec12handle_error(i64 noundef %_0.sroa.0.0.i.ph, i64 %_0.sroa.5.0.i.ph) #13 
           47:  unreachable 
           48:  
           49: bb3: ; preds = %bb12.i 
           50:  %v.0.i = load ptr, ptr %3, align 8, !noalias !2, !nonnull !6, !noundef !6 
           51:  call void @llvm.lifetime.end.p0(i64 24, ptr nonnull %self3.i), !noalias !2 
           52:  store ptr %v.0.i, ptr %slf, align 8, !alias.scope !2 
           53:  %5 = icmp sgt i64 %_0.sroa.0.0.sroa.speculated.i15.i, -1 
           54:  tail call void @llvm.assume(i1 %5) 
           55:  store i64 %_0.sroa.0.0.sroa.speculated.i15.i, ptr %0, align 8, !alias.scope !2 
           56:  ret void 
           57: } 
           58:  
           59: ; <alloc::raw_vec::RawVecInner>::finish_grow 
           60: ; Function Attrs: cold uwtable 
           61: define internal fastcc void @_RNvMs2_NtCsj7oXjtrRIDu_5alloc7raw_vecNtB5_11RawVecInner11finish_growCslAzzQRh6qCt_15append_elements(ptr dead_on_unwind noalias nocapture noundef nonnull writable writeonly align 8 dereferenceable(24) %_0, ptr %self.0.val, i64 %self.8.val, i64 noundef %cap) unnamed_addr #0 { 
           62: start: 
           63:  %_25.i = icmp slt i64 %cap, 0 
           64:  br i1 %_25.i, label %bb10, label %bb13, !prof !9 
           65:  
           66: bb13: ; preds = %start 
           67:  %0 = icmp eq i64 %self.8.val, 0 
           68:  br i1 %0, label %bb7, label %bb15 
           69:  
           70: bb7: ; preds = %bb13 
           71:  %1 = icmp eq i64 %cap, 0 
           72:  br i1 %1, label %bb18, label %bb4.i.i 
           73:  
           74: bb4.i.i: ; preds = %bb7 
           75: ; call __rustc::__rust_no_alloc_shim_is_unstable_v2 
           76:  tail call void @_RNvCslvZcJmHpdlf_7___rustc35___rust_no_alloc_shim_is_unstable_v2() #14 
           77: ; call __rustc::__rust_alloc 
           78:  %2 = tail call noundef ptr @_RNvCslvZcJmHpdlf_7___rustc12___rust_alloc(i64 noundef %cap, i64 noundef range(i64 1, -9223372036854775807) 1) #14 
           79:  br label %bb9 
           80:  
           81: bb15: ; preds = %bb13 
           82:  %3 = icmp ne ptr %self.0.val, null 
           83:  tail call void @llvm.assume(i1 %3) 
           84:  %cond.not.i.i = icmp ult i64 %cap, %self.8.val 
           85:  br i1 %cond.not.i.i, label %bb2.i.i18, label %_RNvXs_NtCsj7oXjtrRIDu_5alloc5allocNtB4_6GlobalNtNtCsk5lId1djSsD_4core5alloc9Allocator4grow.exit, !prof !10 
           86:  
           87: bb2.i.i18: ; preds = %bb15 
           88: ; call core::panicking::panic_fmt 
           89:  tail call void @_RNvNtCsk5lId1djSsD_4core9panicking9panic_fmt(ptr noundef nonnull inttoptr (i64 145 to ptr), ptr noundef nonnull @alloc_7e80d81941cf5c819e3db4cff23967f9, ptr noalias noundef nonnull readonly align 8 dereferenceable(32) @alloc_69bc7ceeb9f224ad693d5453068fc3d5) #15 
           90:  unreachable 
           91:  
           92: _RNvXs_NtCsj7oXjtrRIDu_5alloc5allocNtB4_6GlobalNtNtCsk5lId1djSsD_4core5alloc9Allocator4grow.exit: ; preds = %bb15 
           93: ; call __rustc::__rust_realloc 
           94:  %raw_ptr.i.i = tail call noundef ptr @_RNvCslvZcJmHpdlf_7___rustc14___rust_realloc(ptr noundef nonnull %self.0.val, i64 noundef %self.8.val, i64 noundef range(i64 1, -9223372036854775807) 1, i64 noundef %cap) #14 
           95:  br label %bb9 
           96:  
           97: bb9: ; preds = %bb4.i.i, %_RNvXs_NtCsj7oXjtrRIDu_5alloc5allocNtB4_6GlobalNtNtCsk5lId1djSsD_4core5alloc9Allocator4grow.exit 
           98:  %raw_ptr.i.i.pn = phi ptr [ %raw_ptr.i.i, %_RNvXs_NtCsj7oXjtrRIDu_5alloc5allocNtB4_6GlobalNtNtCsk5lId1djSsD_4core5alloc9Allocator4grow.exit ], [ %2, %bb4.i.i ] 
           99:  %4 = icmp eq ptr %raw_ptr.i.i.pn, null 
          100:  br i1 %4, label %bb17, label %bb18 
          101:  
          102: bb17: ; preds = %bb9 
          103:  %5 = getelementptr inbounds nuw i8, ptr %_0, i64 8 
          104:  store i64 1, ptr %5, align 8 
          105:  br label %bb10 
          106:  
          107: bb18: ; preds = %bb7, %bb9 
          108:  %raw_ptr.i.i.pn8 = phi ptr [ %raw_ptr.i.i.pn, %bb9 ], [ inttoptr (i64 1 to ptr), %bb7 ] 
          109:  %6 = getelementptr inbounds nuw i8, ptr %_0, i64 8 
          110:  store ptr %raw_ptr.i.i.pn8, ptr %6, align 8 
          111:  br label %bb10 
          112:  
          113: bb10: ; preds = %start, %bb18, %bb17 
          114:  %.sink10 = phi i64 [ 16, %bb18 ], [ 16, %bb17 ], [ 8, %start ] 
          115:  %cap.sink = phi i64 [ %cap, %bb18 ], [ %cap, %bb17 ], [ 0, %start ] 
          116:  %.sink = phi i64 [ 0, %bb18 ], [ 1, %bb17 ], [ 1, %start ] 
          117:  %7 = getelementptr inbounds nuw i8, ptr %_0, i64 %.sink10 
          118:  store i64 %cap.sink, ptr %7, align 8 
          119:  store i64 %.sink, ptr %_0, align 8 
          120:  ret void 
          121: } 
          122:  
          123: ; Function Attrs: uwtable 
          124: define void @vec_append_with_temp_alloc(ptr noalias nocapture noundef align 8 dereferenceable(24) %dst, ptr noalias nocapture noundef nonnull readonly align 1 %src.0, i64 noundef range(i64 0, -9223372036854775808) %src.1) unnamed_addr #1 personality ptr @rust_eh_personality { 
check:12'0                                            X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: no match found
          125: start: 
check:12'0     ~~~~~~~
          126:  %0 = icmp eq i64 %src.1, 0 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          127:  br i1 %0, label %_RINvNtCsk5lId1djSsD_4core3ptr13drop_in_placeINtNtCsj7oXjtrRIDu_5alloc3vec3VechEECslAzzQRh6qCt_15append_elements.exit7, label %bb2.i 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          128:  
check:12'0     ~
          129: bb2.i: ; preds = %start 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~
          130: ; call __rustc::__rust_no_alloc_shim_is_unstable_v2 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          131:  tail call void @_RNvCslvZcJmHpdlf_7___rustc35___rust_no_alloc_shim_is_unstable_v2() #14, !noalias !11 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          132: ; call __rustc::__rust_alloc 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:12'1     ?                             possible intended match
          133:  %1 = tail call noundef ptr @_RNvCslvZcJmHpdlf_7___rustc12___rust_alloc(i64 noundef range(i64 1, -9223372036854775808) %src.1, i64 noundef range(i64 1, -9223372036854775807) 1) #14, !noalias !11 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          134:  %2 = icmp eq ptr %1, null 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
          135:  br i1 %2, label %bb3.i.i, label %bb2.i5, !prof !10 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          136:  
check:12'0     ~
          137: bb3.i.i: ; preds = %bb2.i 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
          138: ; call alloc::raw_vec::handle_error 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          139:  tail call void @_RNvNtCsj7oXjtrRIDu_5alloc7raw_vec12handle_error(i64 noundef 1, i64 range(i64 1, -9223372036854775808) %src.1) #13, !noalias !17 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          140:  unreachable 
check:12'0     ~~~~~~~~~~~~~
          141:  
check:12'0     ~
          142: bb2.i5: ; preds = %bb2.i 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~
          143:  tail call void @llvm.memcpy.p0.p0.i64(ptr nonnull align 1 %1, ptr nonnull readonly align 1 %src.0, i64 range(i64 0, -9223372036854775808) %src.1, i1 false), !noalias !18 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          144:  tail call void @llvm.experimental.noalias.scope.decl(metadata !19) 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          145:  %3 = getelementptr inbounds nuw i8, ptr %dst, i64 16 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          146:  %len.i.i = load i64, ptr %3, align 8, !alias.scope !22, !noundef !6 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          147:  %4 = getelementptr inbounds nuw i8, ptr %dst, i64 8 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          148:  %self2.i.i = load i64, ptr %4, align 8, !range !5, !alias.scope !22, !noundef !6 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          149:  %_9.i.i = sub i64 %self2.i.i, %len.i.i 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          150:  %_7.i.i = icmp ugt i64 %src.1, %_9.i.i 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          151:  br i1 %_7.i.i, label %bb1.i.i, label %_RNvXs_NtCsj7oXjtrRIDu_5alloc5allocNtB4_6GlobalNtNtCsk5lId1djSsD_4core5alloc9Allocator10deallocate.exit.i.i.i4.i6, !prof !10 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          152:  
check:12'0     ~
          153: bb1.i.i: ; preds = %bb2.i5 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
          154: ; invoke <alloc::raw_vec::RawVecInner<_>>::reserve::do_reserve_and_handle::<alloc::alloc::Global> 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          155:  invoke fastcc void @_RINvNvMs2_NtCsj7oXjtrRIDu_5alloc7raw_vecINtB8_11RawVecInnerpE7reserve21do_reserve_and_handleNtNtBa_5alloc6GlobalECslAzzQRh6qCt_15append_elements(ptr noalias noundef nonnull align 8 dereferenceable(24) %dst, i64 noundef %len.i.i, i64 noundef range(i64 1, 0) %src.1) 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          156:  to label %.noexc unwind label %bb3 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          157:  
check:12'0     ~
          158: .noexc: ; preds = %bb1.i.i 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
          159:  %len.pre.i = load i64, ptr %3, align 8, !alias.scope !19 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          160:  %self1.pre.i = load i64, ptr %4, align 8, !range !5, !alias.scope !19 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          161:  br label %_RNvXs_NtCsj7oXjtrRIDu_5alloc5allocNtB4_6GlobalNtNtCsk5lId1djSsD_4core5alloc9Allocator10deallocate.exit.i.i.i4.i6 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          162:  
check:12'0     ~
          163: _RNvXs_NtCsj7oXjtrRIDu_5alloc5allocNtB4_6GlobalNtNtCsk5lId1djSsD_4core5alloc9Allocator10deallocate.exit.i.i.i4.i6: ; preds = %.noexc, %bb2.i5 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          164:  %self1.i = phi i64 [ %self2.i.i, %bb2.i5 ], [ %self1.pre.i, %.noexc ] 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          165:  %len.i = phi i64 [ %len.i.i, %bb2.i5 ], [ %len.pre.i, %.noexc ] 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          166:  %_12.i = icmp sgt i64 %len.i, -1 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          167:  tail call void @llvm.assume(i1 %_12.i) 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          168:  %cond.i = icmp ne i64 %self1.i, 0 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          169:  tail call void @llvm.assume(i1 %cond.i) 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          170:  %_17.i = load ptr, ptr %dst, align 8, !alias.scope !19, !nonnull !6, !noundef !6 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          171:  %dst.i = getelementptr inbounds nuw i8, ptr %_17.i, i64 %len.i 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          172:  tail call void @llvm.memcpy.p0.p0.i64(ptr nonnull align 1 %dst.i, ptr nonnull readonly align 1 %1, i64 %src.1, i1 false), !noalias !19 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          173:  %_11.0.i = add nuw i64 %len.i, %src.1 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          174:  store i64 %_11.0.i, ptr %3, align 8, !alias.scope !19 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          175: ; call __rustc::__rust_dealloc 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          176:  tail call void @_RNvCslvZcJmHpdlf_7___rustc14___rust_dealloc(ptr noundef nonnull %1, i64 noundef %src.1, i64 noundef range(i64 1, -9223372036854775807) 1) #14 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          177:  br label %_RINvNtCsk5lId1djSsD_4core3ptr13drop_in_placeINtNtCsj7oXjtrRIDu_5alloc3vec3VechEECslAzzQRh6qCt_15append_elements.exit7 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          178:  
check:12'0     ~
          179: _RINvNtCsk5lId1djSsD_4core3ptr13drop_in_placeINtNtCsj7oXjtrRIDu_5alloc3vec3VechEECslAzzQRh6qCt_15append_elements.exit7: ; preds = %start, %_RNvXs_NtCsj7oXjtrRIDu_5alloc5allocNtB4_6GlobalNtNtCsk5lId1djSsD_4core5alloc9Allocator10deallocate.exit.i.i.i4.i6 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          180:  ret void 
check:12'0     ~~~~~~~~~~
          181:  
check:12'0     ~
          182: bb3: ; preds = %bb1.i.i 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~
          183:  %5 = landingpad { ptr, i32 } 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          184:  cleanup 
check:12'0     ~~~~~~~~~
          185: ; call __rustc::__rust_dealloc 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          186:  tail call void @_RNvCslvZcJmHpdlf_7___rustc14___rust_dealloc(ptr noundef nonnull %1, i64 noundef %src.1, i64 noundef range(i64 1, -9223372036854775807) 1) #14 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          187:  resume { ptr, i32 } %5 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~
          188: } 
check:12'0     ~~
          189:  
check:12'0     ~
          190: ; Function Attrs: uwtable 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
          191: define void @string_append_with_temp_alloc(ptr noalias nocapture noundef align 8 dereferenceable(24) %dst, ptr noalias nocapture noundef nonnull readonly align 1 %src.0, i64 noundef %src.1) unnamed_addr #1 personality ptr @rust_eh_personality { 
check:12'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:27'0                                               X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: no match found
          192: start: 
check:27'0     ~~~~~~~
          193:  %0 = icmp eq i64 %src.1, 0 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          194:  br i1 %0, label %_RINvNtCsk5lId1djSsD_4core3ptr13drop_in_placeNtNtCsj7oXjtrRIDu_5alloc6string6StringECslAzzQRh6qCt_15append_elements.exit5, label %bb2.i 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          195:  
check:27'0     ~
          196: bb2.i: ; preds = %start 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~
          197: ; call __rustc::__rust_no_alloc_shim_is_unstable_v2 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          198:  tail call void @_RNvCslvZcJmHpdlf_7___rustc35___rust_no_alloc_shim_is_unstable_v2() #14, !noalias !25 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          199: ; call __rustc::__rust_alloc 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:27'1     ?                             possible intended match
          200:  %1 = tail call noundef ptr @_RNvCslvZcJmHpdlf_7___rustc12___rust_alloc(i64 noundef range(i64 1, -9223372036854775808) %src.1, i64 noundef range(i64 1, -9223372036854775807) 1) #14, !noalias !25 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          201:  %2 = icmp eq ptr %1, null 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
          202:  br i1 %2, label %bb3.i.i, label %bb2.i.i.i, !prof !10 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          203:  
check:27'0     ~
          204: bb3.i.i: ; preds = %bb2.i 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
          205: ; call alloc::raw_vec::handle_error 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          206:  tail call void @_RNvNtCsj7oXjtrRIDu_5alloc7raw_vec12handle_error(i64 noundef 1, i64 range(i64 1, -9223372036854775808) %src.1) #13, !noalias !31 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          207:  unreachable 
check:27'0     ~~~~~~~~~~~~~
          208:  
check:27'0     ~
          209: bb2.i.i.i: ; preds = %bb2.i 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          210:  tail call void @llvm.memcpy.p0.p0.i64(ptr nonnull align 1 %1, ptr nonnull readonly align 1 %src.0, i64 range(i64 0, -9223372036854775808) %src.1, i1 false), !noalias !32 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          211:  tail call void @llvm.experimental.noalias.scope.decl(metadata !33) 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          212:  tail call void @llvm.experimental.noalias.scope.decl(metadata !36) 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          213:  tail call void @llvm.experimental.noalias.scope.decl(metadata !39) 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          214:  %3 = getelementptr inbounds nuw i8, ptr %dst, i64 16 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          215:  %len.i.i.i.i = load i64, ptr %3, align 8, !alias.scope !42, !noalias !45, !noundef !6 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          216:  %4 = getelementptr inbounds nuw i8, ptr %dst, i64 8 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          217:  %self2.i.i.i.i = load i64, ptr %4, align 8, !range !5, !alias.scope !42, !noalias !45, !noundef !6 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          218:  %_9.i.i.i.i = sub i64 %self2.i.i.i.i, %len.i.i.i.i 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          219:  %_7.i.i.i.i = icmp ugt i64 %src.1, %_9.i.i.i.i 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          220:  br i1 %_7.i.i.i.i, label %bb1.i.i.i.i, label %_RNvXs_NtCsj7oXjtrRIDu_5alloc5allocNtB4_6GlobalNtNtCsk5lId1djSsD_4core5alloc9Allocator10deallocate.exit.i.i.i4.i.i4, !prof !10 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          221:  
check:27'0     ~
          222: bb1.i.i.i.i: ; preds = %bb2.i.i.i 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          223: ; invoke <alloc::raw_vec::RawVecInner<_>>::reserve::do_reserve_and_handle::<alloc::alloc::Global> 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          224:  invoke fastcc void @_RINvNvMs2_NtCsj7oXjtrRIDu_5alloc7raw_vecINtB8_11RawVecInnerpE7reserve21do_reserve_and_handleNtNtBa_5alloc6GlobalECslAzzQRh6qCt_15append_elements(ptr noalias noundef nonnull align 8 dereferenceable(24) %dst, i64 noundef %len.i.i.i.i, i64 noundef range(i64 1, -9223372036854775808) %src.1) 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          225:  to label %.noexc unwind label %bb3 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          226:  
check:27'0     ~
          227: .noexc: ; preds = %bb1.i.i.i.i 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          228:  %len.pre.i.i.i = load i64, ptr %3, align 8, !alias.scope !47, !noalias !45 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          229:  %self1.pre.i.i.i = load i64, ptr %4, align 8, !range !5, !alias.scope !47, !noalias !45 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          230:  br label %_RNvXs_NtCsj7oXjtrRIDu_5alloc5allocNtB4_6GlobalNtNtCsk5lId1djSsD_4core5alloc9Allocator10deallocate.exit.i.i.i4.i.i4 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          231:  
check:27'0     ~
          232: _RNvXs_NtCsj7oXjtrRIDu_5alloc5allocNtB4_6GlobalNtNtCsk5lId1djSsD_4core5alloc9Allocator10deallocate.exit.i.i.i4.i.i4: ; preds = %.noexc, %bb2.i.i.i 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          233:  %self1.i1.i.i = phi i64 [ %self2.i.i.i.i, %bb2.i.i.i ], [ %self1.pre.i.i.i, %.noexc ] 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          234:  %len.i.i.i = phi i64 [ %len.i.i.i.i, %bb2.i.i.i ], [ %len.pre.i.i.i, %.noexc ] 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          235:  %_12.i.i.i = icmp sgt i64 %len.i.i.i, -1 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          236:  tail call void @llvm.assume(i1 %_12.i.i.i) 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          237:  %cond.i.i.i = icmp ne i64 %self1.i1.i.i, 0 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          238:  tail call void @llvm.assume(i1 %cond.i.i.i) 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          239:  %_17.i.i.i = load ptr, ptr %dst, align 8, !alias.scope !47, !noalias !45, !nonnull !6, !noundef !6 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          240:  %dst.i.i.i = getelementptr inbounds nuw i8, ptr %_17.i.i.i, i64 %len.i.i.i 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          241:  tail call void @llvm.memcpy.p0.p0.i64(ptr nonnull align 1 %dst.i.i.i, ptr nonnull readonly align 1 %1, i64 range(i64 0, -9223372036854775808) %src.1, i1 false), !noalias !47 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          242:  %_11.0.i.i.i = add nuw i64 %len.i.i.i, %src.1 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          243:  store i64 %_11.0.i.i.i, ptr %3, align 8, !alias.scope !47, !noalias !45 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          244: ; call __rustc::__rust_dealloc 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          245:  tail call void @_RNvCslvZcJmHpdlf_7___rustc14___rust_dealloc(ptr noundef nonnull %1, i64 noundef %src.1, i64 noundef range(i64 1, -9223372036854775807) 1) #14 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          246:  br label %_RINvNtCsk5lId1djSsD_4core3ptr13drop_in_placeNtNtCsj7oXjtrRIDu_5alloc6string6StringECslAzzQRh6qCt_15append_elements.exit5 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          247:  
check:27'0     ~
          248: _RINvNtCsk5lId1djSsD_4core3ptr13drop_in_placeNtNtCsj7oXjtrRIDu_5alloc6string6StringECslAzzQRh6qCt_15append_elements.exit5: ; preds = %start, %_RNvXs_NtCsj7oXjtrRIDu_5alloc5allocNtB4_6GlobalNtNtCsk5lId1djSsD_4core5alloc9Allocator10deallocate.exit.i.i.i4.i.i4 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          249:  ret void 
check:27'0     ~~~~~~~~~~
          250:  
check:27'0     ~
          251: bb3: ; preds = %bb1.i.i.i.i 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          252:  %5 = landingpad { ptr, i32 } 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          253:  cleanup 
check:27'0     ~~~~~~~~~
          254: ; call __rustc::__rust_dealloc 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          255:  tail call void @_RNvCslvZcJmHpdlf_7___rustc14___rust_dealloc(ptr noundef nonnull %1, i64 noundef %src.1, i64 noundef range(i64 1, -9223372036854775807) 1) #14 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          256:  resume { ptr, i32 } %5 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~
          257: } 
check:27'0     ~~
          258:  
check:27'0     ~
          259: ; Function Attrs: nounwind uwtable 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          260: declare noundef range(i32 0, 10) i32 @rust_eh_personality(i32 noundef, i32 noundef, i64 noundef, ptr noundef, ptr noundef) unnamed_addr #2 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          261:  
check:27'0     ~
          262: ; Function Attrs: mustprogress nocallback nofree nosync nounwind willreturn memory(argmem: readwrite) 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          263: declare void @llvm.lifetime.start.p0(i64 immarg, ptr nocapture) #3 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          264:  
check:27'0     ~
          265: ; Function Attrs: mustprogress nocallback nofree nosync nounwind willreturn memory(argmem: readwrite) 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          266: declare void @llvm.lifetime.end.p0(i64 immarg, ptr nocapture) #3 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          267:  
check:27'0     ~
          268: ; alloc::raw_vec::handle_error 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          269: ; Function Attrs: cold minsize noreturn optsize uwtable 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          270: declare void @_RNvNtCsj7oXjtrRIDu_5alloc7raw_vec12handle_error(i64 noundef range(i64 0, -9223372036854775807), i64) unnamed_addr #4 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          271:  
check:27'0     ~
          272: ; Function Attrs: mustprogress nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: write) 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          273: declare void @llvm.assume(i1 noundef) #5 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          274:  
check:27'0     ~
          275: ; Function Attrs: mustprogress nocallback nofree nounwind willreturn memory(argmem: readwrite) 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          276: declare void @llvm.memcpy.p0.p0.i64(ptr noalias nocapture writeonly, ptr noalias nocapture readonly, i64, i1 immarg) #6 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          277:  
check:27'0     ~
          278: ; __rustc::__rust_no_alloc_shim_is_unstable_v2 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          279: ; Function Attrs: nounwind uwtable 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          280: declare void @_RNvCslvZcJmHpdlf_7___rustc35___rust_no_alloc_shim_is_unstable_v2() unnamed_addr #2 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          281:  
check:27'0     ~
          282: ; __rustc::__rust_alloc 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~
          283: ; Function Attrs: nounwind allockind("alloc,uninitialized,aligned") allocsize(0) uwtable 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          284: declare noalias noundef ptr @_RNvCslvZcJmHpdlf_7___rustc12___rust_alloc(i64 noundef, i64 allocalign noundef) unnamed_addr #7 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          285:  
check:27'0     ~
          286: ; core::panicking::panic_fmt 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          287: ; Function Attrs: cold noinline noreturn uwtable 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          288: declare void @_RNvNtCsk5lId1djSsD_4core9panicking9panic_fmt(ptr noundef nonnull, ptr noundef nonnull, ptr noalias noundef readonly align 8 dereferenceable(32)) unnamed_addr #8 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          289:  
check:27'0     ~
          290: ; __rustc::__rust_dealloc 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
          291: ; Function Attrs: nounwind allockind("free") uwtable 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          292: declare void @_RNvCslvZcJmHpdlf_7___rustc14___rust_dealloc(ptr allocptr noundef, i64 noundef, i64 noundef) unnamed_addr #9 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          293:  
check:27'0     ~
          294: ; __rustc::__rust_realloc 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
          295: ; Function Attrs: nounwind allockind("realloc,aligned") allocsize(3) uwtable 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          296: declare noalias noundef ptr @_RNvCslvZcJmHpdlf_7___rustc14___rust_realloc(ptr allocptr noundef, i64 noundef, i64 allocalign noundef, i64 noundef) unnamed_addr #10 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          297:  
check:27'0     ~
          298: ; Function Attrs: nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: readwrite) 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          299: declare void @llvm.experimental.noalias.scope.decl(metadata) #11 
check:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            .
            .
            .
>>>>>>

------------------------------------------

error: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-20/bin/FileCheck" "--input-file" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/lib-optimizations/append-elements/append-elements.ll" "/checkout/tests/codegen-llvm/lib-optimizations/append-elements.rs" "--check-prefix=CHECK" "--allow-unused-prefixes" "--dump-input-context" "100"
stdout: none
```

</details>

--- a/compiler/rustc_codegen_llvm/src/attributes.rs
+++ b/compiler/rustc_codegen_llvm/src/attributes.rs
@@ -517,7 +517,16 @@ pub(crate) fn llfn_attrs_from_instance<'ll, 'tcx>(
         to_add.push(llvm::CreateAllocKindAttr(cx.llcx, AllocKindFlags::Free));
         // applies to argument place instead of function place
         let allocated_pointer = AttributeKind::AllocatedPointer.create_attr(cx.llcx);
-        attributes::apply_to_llfn(llfn, AttributePlace::Argument(0), &[allocated_pointer]);
+        let attrs: &[_] = if llvm_util::get_version() >= (21, 0, 0) {
+            // "Does not capture provenance" means "if the function call stashes the pointer somewhere,
+            // accessing that pointer after the function returns is UB". That is definitely the case here since
+            // freeing will destroy the provenance.
+            let captures_addr = AttributeKind::CapturesAddress.create_attr(cx.llcx);
+            &[allocated_pointer, captures_addr]
+        } else {
+            &[allocated_pointer]
+        };
+        attributes::apply_to_llfn(llfn, AttributePlace::Argument(0), attrs);
     }
     if let Some(align) = codegen_fn_attrs.alignment {
         llvm::set_alignment(llfn, align);

--- a/library/alloc/src/slice.rs
+++ b/library/alloc/src/slice.rs
@@ -448,9 +448,11 @@ impl<T> [T] {
                 // SAFETY:
                 // allocated above with the capacity of `s`, and initialize to `s.len()` in
                 // ptr::copy_to_non_overlapping below.
-                unsafe {
-                    s.as_ptr().copy_to_nonoverlapping(v.as_mut_ptr(), s.len());
-                    v.set_len(s.len());
+                if s.len() > 0 {
+                    unsafe {
+                        s.as_ptr().copy_to_nonoverlapping(v.as_mut_ptr(), s.len());
+                        v.set_len(s.len());
+                    }
                 }
                 v
             }

--- a/library/alloc/src/vec/mod.rs
+++ b/library/alloc/src/vec/mod.rs
@@ -2818,7 +2818,11 @@ impl<T, A: Allocator> Vec<T, A> {
         let count = other.len();
         self.reserve(count);
         let len = self.len();
-        unsafe { ptr::copy_nonoverlapping(other as *const T, self.as_mut_ptr().add(len), count) };
+        if count > 0 {
+            unsafe {
+                ptr::copy_nonoverlapping(other as *const T, self.as_mut_ptr().add(len), count)
+            };
+        }
         self.len += count;
     }
 

--- a/tests/codegen-llvm/lib-optimizations/append-elements.rs
+++ b/tests/codegen-llvm/lib-optimizations/append-elements.rs
@@ -1,0 +1,30 @@
+//@ compile-flags: -O -Zmerge-functions=disabled
+//@ min-llvm-version: 21
+#![crate_type = "lib"]
+
+//! Check that a temporary intermediate allocations can eliminated and replaced
+//! with memcpy forwarding.
+//! This requires Vec code to be structured in a way that avoids phi nodes from the
+//! zero-capacity length flowing into the memcpy arguments.
+
+// CHECK-LABEL: @vec_append_with_temp_alloc
+#[no_mangle]
+pub fn vec_append_with_temp_alloc(dst: &mut Vec<u8>, src: &[u8]) {
+    // CHECK-NOT: call void @llvm.memcpy
+    // CHECK: call void @llvm.memcpy.{{.*}}%dst.i{{.*}}%src.0
+    // CHECK-NOT: call void @llvm.memcpy
+    let temp = src.to_vec();
+    dst.extend(&temp);
+    // CHECK: ret
+}
+
+// CHECK-LABEL: @string_append_with_temp_alloc
+#[no_mangle]
+pub fn string_append_with_temp_alloc(dst: &mut String, src: &str) {
+    // CHECK-NOT: call void @llvm.memcpy
+    // CHECK: call void @llvm.memcpy.{{.*}}%dst.i{{.*}}%src.0
+    // CHECK-NOT: call void @llvm.memcpy
+    let temp = src.to_string();
+    dst.push_str(&temp);
+    // CHECK: ret
+}

